File: src/test/java/io/icednut/algorithm/exercise/graph/leet733/SolutionTest.java
package io.icednut.algorithm.exercise.graph.leet733;

import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.assertArrayEquals;

class SolutionTest {

    @Test
    void test01() {
        var solution = new Solution();

        int[][] output = solution.floodFill(
                new int[][]{
                        new int[]{1, 1, 1},
                        new int[]{1, 1, 0},
                        new int[]{1, 0, 1},
                },
                1, 1, 2
        );

        assertArrayEquals(
                new int[][]{
                        new int[]{2, 2, 2},
                        new int[]{2, 2, 0},
                        new int[]{2, 0, 1},
                },
                output
        );
    }

    @Test
    void test02() {
        var solution = new Solution();

        int[][] output = solution.floodFill(
                new int[][]{
                        new int[]{0, 0, 0},
                        new int[]{0, 0, 0},
                        new int[]{0, 0, 0},
                },
                0, 0, 0
        );

        assertArrayEquals(
                new int[][]{
                        new int[]{0, 0, 0},
                        new int[]{0, 0, 0},
                        new int[]{0, 0, 0},
                },
                output
        );
    }
}

File: src/test/kotlin/io/icednut/algorithm/exercise/graph/leet733/SolutionTest.kt
package io.icednut.algorithm.exercise.graph.leet733

import org.junit.jupiter.api.Assertions.*
import org.junit.jupiter.api.Test

class SolutionTest {

    @Test
    fun test01() {
        val solution = Solution()

        val floodFill = solution.floodFill(
            image = arrayOf(
                intArrayOf(1, 1, 1),
                intArrayOf(1, 1, 0),
                intArrayOf(1, 0, 1),
            ),
            sr = 1,
            sc = 1,
            color = 2
        )

        assertArrayEquals(
            arrayOf(
                intArrayOf(2, 2, 2),
                intArrayOf(2, 2, 0),
                intArrayOf(2, 0, 1),
            ),
            floodFill
        )
    }

    @Test
    fun test02() {
        val solution = Solution()

        val floodFill = solution.floodFill(
            image = arrayOf(
                intArrayOf(0, 0, 0),
                intArrayOf(0, 0, 0)
            ),
            sr = 0,
            sc = 0,
            color = 0
        )

        assertArrayEquals(
            arrayOf(
                intArrayOf(0, 0, 0),
                intArrayOf(0, 0, 0)
            ),
            floodFill
        )
    }

    @Test
    fun test03() {
        val solution = Solution()

        val floodFill = solution.floodFill(
            image = arrayOf(
                intArrayOf(0, 0, 0),
                intArrayOf(0, 0, 0)
            ),
            sr = 1,
            sc = 0,
            color = 2
        )

        assertArrayEquals(
            arrayOf(
                intArrayOf(2, 2, 2),
                intArrayOf(2, 2, 2)
            ),
            floodFill
        )
    }
}